packetbeat/_meta/fields.yml

```diff
@@ -1748,3 +1748,293 @@
+              type: array
+              description: >
+                List of ciphers the client is willing to use for this session.
+                See https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4
+
+            - name: supported_compression_methods
+              type: array
+              description: >
+                The list of compression methods the client supports.
+                See https://www.iana.org/assignments/comp-meth-ids/comp-meth-ids.xhtml
+
+            - name: extensions
+              type: group
+              description: The hello extensions provided by the client.
+              fields:
+                - name: server_name_indication
+                  type: string
+                  description: Comma-separated list of server names
+
+                - name: application_layer_protocol_negotiation
+                  type: string
+                  description: Comma-separated list of protocols
+
+                - name: session_ticket
+                  type: string
+                  description: >
+                    Length of the session ticket, if provided, or empty
+
+        - name: server_hello
+          type: group
+          fields:
+            - name: version
+              type: keyword
+              description: >
+                The version of the TLS protocol that is used for this session.
+                It is the highest version supported by the server not exceeding
+                the version requested in the client hello.
+
+            - name: timestamp
+              type: long
+              description: >
+                The current time and date in standard UNIX 32-bit format
+                according to the server's internal clock.
+
+            - name: selected_cipher
+              type: keyword
+              description: >
+                The cipher suite selected by the server from the list provided
+                by in the client hello.
+
+            - name: selected_compression_method
+              type: keyword
+              description: >
+                The compression method selected by the server from the list
+                provided in the client hello.
+
+            - name: extensions
+              type: group
+              description: The hello extensions provided by the server.
+              fields:
+                - name: application_layer_protocol_negotiation
+                  type: string
+                  description: Negotiated application layer protocol
+
+                - name: session_ticket
+                  type: string
+                  description: Empty
+
+        - name: client_certificate
+          type: group
+          description: Certificate provided by the client for authentication.
+          fields:
+
+            - name: version
+              type: long
+              description: X509 format version.
+
+            - name: serial_number
+              type: keyword
```

Comment:
Are the some benefits to sending this as a keyword rather than a long? Will users be expecting to see hex? Maybe we should have a field formater in Kibana for hex (possibly there is one; I didn’t check).

Reply:
The serial number can be up to 160 bits long so it doesn't fit on a numeric type.

I have changed the format to decimal. The most common  way of representation seems to be decimal up to 64 bit in length and then hexadecimal number or hex dump of individual bytes. I will stick with decimal in a keyword field.